core/server/master/src/main/java/alluxio/master/file/meta/LockedInodePath.java

```diff
@@ -450,8 +450,8 @@ public void traverse() throws InvalidPathException {
           }
         }
         if (!nextInodeOpt.isPresent()) {
-          if (!mLockPattern.writeLockLastEdge()) {
-            // Other lock patterns only lock up to the last existing inode.
+          if (mLockPattern == LockPattern.READ) {
+            // Other lock patterns should lock the last non-existing edge.
```

Comment:
"other" means WRITE_INODE and WRITE_EDGE? Maybe we should be a bit more explicit with this comment for clarity?